### PR TITLE
nyxt: 2020-10-23 -> 2021-03-27

### PR DESCRIPTION
**Pull request by 7c6f434c**
###### Motivation for this change

Nyxt was too old, some changes needed to build a newer one

Will eventually self-merge

@atlas-engineer @nlewo @bqv 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117902` at c10b36b0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>lispPackages.cl-custom-hash-table</li>
    <li>lispPackages.nyxt</li>
    <li>nyxt</li>
  </ul>
</details>

File: pkgs/development/lisp-modules/lisp-packages.nix [hunk 1]
```diff
@@ -143,59 +142,56 @@ let lispPackages = rec {
...
+                    trivial-features
+                    trivial-package-local-nicknames
+                    trivial-types
+                    unix-opts
+                    cl-html-diff
+                    hu_dot_dwim_dot_defclass-star
+                    cl-custom-hash-table
+                    fset
+                    cl-cffi-gtk
+                    cl-webkit2
```
Comment (SuperSandro2000) [inline pkgs/development/lisp-modules/lisp-packages.nix:183]: Please fix this formatting.

File: pkgs/development/lisp-modules/quicklisp-to-nix-output/cl-custom-hash-table.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
+args @ { fetchurl, ... }:
+rec {
+  baseName = ''cl-custom-hash-table'';
+  version = ''20201220-git'';
```
Comment (SuperSandro2000) [inline pkgs/development/lisp-modules/quicklisp-to-nix-output/cl-custom-hash-table.nix:4]: Please use " for single line strings.

File: pkgs/development/lisp-modules/lisp-packages.nix (hunk 1 above)
Comment (7c6f434c) [reply pkgs/development/lisp-modules/lisp-packages.nix:183]: Oops, wrong autoindent applied, sorry

File: pkgs/development/lisp-modules/quicklisp-to-nix-output/cl-custom-hash-table.nix (hunk 2 above)
Comment (7c6f434c) [reply pkgs/development/lisp-modules/quicklisp-to-nix-output/cl-custom-hash-table.nix]: Hm, the change creates only one case of escape use in the quicklisp-to-nix output, so can as well regenerate.

**Comment by jmercouris** [comment]
Really excited for this!

**Comment by Ambrevar** [comment]
Thanks a lot, it's fantastic to see Nyxt in Nix again! :D

Note: Osicat should no longer be necessary if you build Nyxt with SBCL.

**Comment by 7c6f434c** [comment]
>Thanks a lot, it's fantastic to see Nyxt in Nix again! :D

Well, Nyxt _was_ there, just too out of date (although still less out of date than the upstream release, so yeah)

>Note: Osicat should no longer be necessary if you build Nyxt with SBCL.

I noticed that #- but decided to just keep it in (maybe it even works with CCL? I haven't tried)

Given the huge WebKit FFI-dependency, minor stuff like osicat does not look like worth optimising, especially with subsystem dependency lists hidden deeper inside etc.

**Comment by Ambrevar** [comment]
> I noticed that #- but decided to just keep it in (maybe it even works with CCL? I haven't tried)

It works with CCL indeed, but there are issues when dumping the image.
I need to work it out.

> Given the huge WebKit FFI-dependency, minor stuff like osicat does not look like worth optimising, especially with subsystem dependency lists hidden deeper inside etc.

It's not just for size optimization but rather because osicat is a
tricky dependency to work with because it tries to compile files to its
directory when it's loaded.
This can be avoided if the package is carefully built.  Just so you
know: if you run into issues with Osicat, feel free to remove it from
the Nyxt deps.

**Comment by jmercouris** [comment]
Thanks a lot!

**Comment by 7c6f434c** [comment]
>> Given the huge WebKit FFI-dependency, minor stuff like osicat does not look like worth optimising, especially with subsystem dependency lists hidden deeper inside etc.
>
>It's not just for size optimization but rather because osicat is a
>tricky dependency to work with because it tries to compile files to its
>directory when it's loaded.

That part about compiling a .so into its directory is actually easy (and I need it to work for CL-Fuse / QueryFS anyway, so I will keep it working in Nixpkgs…). Funnily enough, there is some issue on CCL that it tries to force recompiling a part of trivial-features (works fine with SBCL, though).

**Comment by tg-x** [comment]
nice to see an effort of getting nyxt build on nix!

just tried this build via `nix run 'nixpkgs#nyxt'` from the latest nixpkgs master

first two issues i found:
- no TLS support available (TLS/SSL support not available; install glib-networking)
- fonts are too small and `zoom-in-page` doesnt increase them, only the rest of the page

**Comment by 7c6f434c** [comment]
>nice to see an effort of getting nyxt build on nix!
>
>just tried this build via `nix run 'nixpkgs#nyxt'` from the latest nixpkgs master

I have no idea how many bugs there are in the choice of «which Nyxt» here…

>first two issues i found:
>- no TLS support available (TLS/SSL support not available; install glib-networking)

Under a fresh user works fine for me. This _does_ happen if I run «lispPackages.nyxt» version instead of top-level «nyxt» version, though

>- fonts are too small and `zoom-in-page` doesnt increase them, only the rest of the page

Maybe the page uses some creative styling that WebKit refuses to scale? Does Midori behave the same?

**Comment by tg-x** [comment]
I would expect that was the top-level nyxt running, what's the difference though?

As for the font issue, that happened already on the nyxt help page (M-x help), which worked fine on the nyxt guix pack I downloaded before.

**Comment by 7c6f434c** [comment]
>I would expect that was the top-level nyxt running, what's the difference though?

The top-level application Nyxt is wrapped with all the GTK environment setup; lispPackages part is not.

>As for the font issue, that happened already on the nyxt help page (M-x help), which worked fine on the nyxt guix pack I downloaded before.

Works for me. Also, isn't it C-space help instead of M-x help?

Have you tried it under a freshly-created empty user account?

(My testing was in a pretty restricted jail with empty but writeable $HOME)

**Comment by tg-x** [comment]
OK the TLS issue is resolved now, not sure why it didn't work the first time, I'm running it the same way the top-level nyxt.

The font issue is still there, though, perhaps it needs some extra env vars or font-config to find the fonts?

**Comment by 7c6f434c** [comment]
Oh, I have an Xorg-only setup, I guess that's why I am not seeing the font issue.

Not even sure at what level the problem is — font scaling is fine in Midori with Wayland, right?

**Comment by tg-x** [comment]
fonts in midori work fine both with wayland and x11
(the toolbar icons are missing though in both cases)

**Comment by 7c6f434c** [comment]
Then I am lost… Not sure which part should behave differently between X11 and GTK and not be used by Midori.

**Pull request merged**